R/rintrinio.R

```diff
@@ -56,8 +57,35 @@ gather_financial_statement_company_compare <- function(api_key, ticker, statemen
 #'
 #' @examples
 #' gather_stock_time_series(api_key, 'AAPL', "2017-12-31", "2019-03-01")
-gather_stock_time_series <- function(api_key, ticker, start_date, end_date) {
-  tibble()
+gather_stock_time_series <- function(api_key, ticker, start_date='', end_date='') {
+  
+  # set up options
+  if (start_date != '' & end_date != '') {
+    opts <- list(start_date = as.Date(start_date), end_date = as.Date(end_date), page_size = 10000)
+  }
+  else if (start_date != '') {
+    opts <- list(start_date = as.Date(start_date), page_size = 10000)
+  }
+  else if (end_date != '') {
+    opts <- list(end_date = as.Date(end_date), page_size = 10000)
+  }
+  else {
+    opts <- list(page_size = 10000)
+  }
```

Comment:
It will be nice if we could just involve all the exception handling parts as we did in the python version. (`date` format, `date` sequence, `ticker` format etc. )  
Attaching my code just for your reference : 
```
t <- try({buy_date <- as.Date(buy_date)
         sell_date <- as.Date(sell_date)}, silent=T)
if("try-error" %in% class(t)) {
    return("Error message: xxxxxx.")
}
```

Reply:
I had a lot of issues with the code for this.....it actually kept breaking the full function so I am not going to implement this now but opened #43 in hopes to address this.